packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx

```diff
@@ -32,6 +32,16 @@ export const GatsbyImage: FunctionComponent<GatsbyImageProps> = function GatsbyI
     console.warn(`[gatsby-plugin-image] Missing image prop`)
     return null
   }
+  if (`class` in props) {
+    className = props.class
+    delete props.class
+  }
```

Comment:
Same comment as before; add `class: className2` to function param so it doesn't become part of the `props` rest param. Then you can do a simple `if (className)` check and won't have to `delete`.

Maybe you can even hack around the duplicate name through inits but I wouldn't recommend.